### ISSUE TITLE
incompact container name between `docker-compose` and `podman-compose`

### ISSUE THREAD

**Issue by loynoir** [containers/podman-compose#588]
**Describe the bug**
incompact container name between `docker-compose` and `podman-compose`

**To Reproduce**
`foobar/docker-compose.yml`
```
version: '2'
services:
  hello_world:
    image: ubuntu
    command: [/bin/echo, 'Hello world']
```

```
$ cd foobar
$ docker-compose up
$ BUILDAH_FORMAT=docker PODMAN_USERNS=keep-id podman-compose up
```

**Expected behavior**
```
$ docker container ls -a
... NAMES
... foobar-hello_world-1
```

**Actual behavior**
```
$ podman container ls -a
... NAMES
... foobar_hello_world_1
```

**Output**

```
$ podman-compose version
using podman version: 4.3.0
podman-composer version  1.0.3
podman --version
podman version 4.3.0

$ podman-compose up
...

```

**Environment:**
 - OS: Linux / WSL / Mac
 - podman version: 
 - podman compose version: (git hex)

**Additional context**

**Comment by benyaminl** [comment]
could you just use 
`podman-compose exec -it hello_world bash -c '/bin/foo'` 

Or is there specific problem that needs it for now?

**Comment by loynoir** [comment]
@benyaminl

> Or is there specific problem that needs it for now?

Yes, maybe.

Although I don't have clue that two issues are related, 

this specific difference remind me of https://github.com/microsoft/vscode-dev-containers/issues/1595 , 

which actually use `reproduce_devcontainer_app` instead of expected `reproduce_devcontainer-app`, 

so devcontainer fail to start when using docker-compose.

**Comment by loynoir** [comment]
@benyaminl 
- Above code is just for reproduce.
- Specific problem? Yes, there was a specific problem. My automate script failed. Because of different name between `podman-compose` and `docker-compose`
- At the day I reply this message, `podman-compose` still create container name with `foobar_hello_world_1` instead of `docker-compose` style `foobar-hello_world_1`

**Comment by benyaminl** [comment]
@loynoir for some reason, mine on alpine and fedora comply with the docker-compose format, it's strange tho.

I'm using version 1.0.4 not 1.0.3

**Comment by davidfstr** [comment]
+1: The incompatible container name format when comparing `docker-compose` with `podman-compose` (i.e. `my-website-app-1` vs `my-website_app_1`) **made my migration from Docker to Podman more difficult**. It would be great if podman-compose's name format was at least configurable to match the docker-compose format even if it doesn't use that name format by default.

For almost every part of the migration I was able to symlink `docker` -> `podman` and get close enough behavior that I didn't need to simultaneously update all my build scripts that were using `docker`. However I _did_ need to make some scripts aware that podman-compose was being used rather than docker-compose to accommodate the incompatible container name format

**Comment by teohhanhui** [comment]
@davidfstr https://github.com/containers/podman-compose/blob/main/docs/Extensions.md#docker-compose-compatibility

TL;DR: Set `PODMAN_COMPOSE_DOCKER_COMPOSE_COMPAT=1` in your shell's environment.

**Comment by davidfstr** [comment]
Thanks @teohhanhui !

Perhaps this bug can actually be closed then, unless it's being kept open until `PODMAN_COMPOSE_DOCKER_COMPOSE_COMPAT=1` becomes the default.
